qa/tasks/cephfs/test_volumes.py

```diff
@@ -4345,6 +4354,407 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
+        changed to false, the async purging of a subvolume will resume and
+        also finish, causing trash to be empty.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self._do_subvolume_io(sv, number_of_files=100)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self.config_set('mgr', self.conf_opt, True)
+        time.sleep(2)
+
+        # XXX actual test here: test that purging is resumed and finished
+        self.config_set('mgr', self.conf_opt, False)
+        self._wait_for_trash_empty()
+
+    def test_the_default_value(self):
+        '''
+        Test that MGR config option mgr/volumes/pause_purging is false by
+        by default.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+
+        o = self.get_ceph_cmd_stdout(f'config get mgr {self.conf_opt}')
+        o = o.lower().strip()
+        self.assertEqual(o, 'false')
+
+    def test_that_subvols_are_purged_by_default(self):
+        '''
+        Test that by default deleting/trashing a subvolume also purge it.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self._do_subvolume_io(sv, number_of_files=10)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self._wait_for_trash_empty()
+
+
+class TestPauseCloning(TestVolumesHelper):
+    '''
+    Tests related to config "mgr/volumes/pause_cloning".
+    '''
+
+    CLIENTS_REQUIRED = 1
+    MDSS_REQUIRED = 1
+
+    conf_opt = 'mgr/volumes/pause_cloning'
+
+    def setUp(self):
+        super().setUp()
+
+        self.NUM_OF_CLONER_THREADS = 4
+        self.config_set('mgr', 'mgr/volumes/max_concurrent_clones', self.NUM_OF_CLONER_THREADS)
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_no_wait', 'false')
+
+    def tearDown(self):
+        # every test will change value of this config option as per its need.
+        # assure that this config option's default value is re-stored during
+        # tearDown() so that there's zero chance that it interferes with next
+        # test.
+        self.config_set('mgr', self.conf_opt, False)
+
+        # ensure purge threads have no jobs left from previous test so that
+        # next test doesn't have to face unnecessary complications.
+        self._wait_for_trash_empty()
+
+        super().tearDown()
+
+    def test_the_default_value(self):
+        '''
+        Test that MGR config option mgr/volumes/pause_cloning is false by
+        by default.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+
+        o = self.get_ceph_cmd_stdout(f'config get mgr {self.conf_opt}')
+        o = o.lower().strip()
+        self.assertEqual(o, 'false')
+
+    def test_when_true_cloning_never_begins(self):
```

Comment:
descriptions missing in the tests cases. test name doesn't tell which config is true ?

Reply:
> test name doesn't tell which config is true

It doesn't need to, class name does that. All tests under `TestPausePurging` are to `pause_purging` and all tests under `TestPauseCloning` are for `pause_cloninig`.